lib/vtls/openssl.c

```diff
@@ -154,8 +154,16 @@ static unsigned long OpenSSL_version_num(void)
 #define OSSL_PACKAGE "OpenSSL"
 #endif
 
+#if (OPENSSL_VERSION_NUMBER >= 0x10002000L)
```

Comment:
I wouldn't change it for at least for OpenSSL 1.0.2, which includes RC4 MD5 support and maybe other things depending on how it was built. By default our current cipher settings disable RC4, but this would change that so -1 for that specific thing.

The commit referred to in that thread you quoted probably concerns just 1.1, and it does say RC4 is not part of the default cipher list when built in. So my suggestion is do it for OpenSSL 1.1 and above. The only other thing is @STRENGTH, I really don't know how important that is for 1.1.

Reply:
Oh right, the `@STRENGTH`. I added that back in commit 0d1060f21efe as a sort of work-around for HTTP/2 since back then OpenSSL would otherwise happily negotiate using ciphers that are deemed "illegal" for HTTP/2 ...

Reply:
Using OPENSSL_VERSION_NUMBER >= 0x10100000L should be of course OK. However I do not see how using @STRENGTH would help to eliminate "illegal" ciphers. This just sorts the ciphersuite list so all AES256 ciphers (including the CBC ones) would come before the AES128 ciphers. I do not think this is right for most use cases. Also the black list of the ciphers should apply mostly to the HTTP/2 servers - the clients are allowed to include them.

Reply:
Still, `@STRENGTH` _did_ help for HTTP/2 and that's why we have it there. Unfortunately I can't recall the picky servers I tried against back then so I can't really check right now what happens against them if/when we drop it.

Reply:
Also you did not make the experiment with OpenSSL 1.1, did you?

Reply:
This is the actual difference between curl(-) and upstream default(+)  https://paste.fedoraproject.org/paste/XPMGLMTvd4d-BI9~CYU1tw

Reply:
This is of course with OpenSSL 1.1.0 - to me the upstream default looks like more sane.